src/app/components/tree/tree.component.ts

```diff
@@ -94,6 +100,21 @@ export class TreeComponent implements AfterContentInit, OnDestroy {
   ngOnDestroy() { // PrimeNG as of 6.0 has no native right click support for its tree
     this.fileExplorerTree.nativeElement.removeEventListener('contextmenu', this.panelRightClickSelect.bind(this));
   }
+
+  onDrop(_event?: any) { 
+    this.dragAndDropEvent.emit({dragData: _event.dragNode, dropData: _event.dropNode});
+  }
+
+  _overrideAllowDrop(dragNode, dropNode, dragNodeScope): boolean {
+    if (dragNode && dropNode) {
+      if(dragNode.data == 'Folder' || dropNode.data != 'Folder' || dragNode.parent.path === dropNode.path) {
```

Comment:
This gives me run-time issues with
externals.js:2 ERROR TypeError: Cannot read properties of undefined (reading 'path')
    at e._overrideAllowDrop (tree.component.ts:110:84)
    at s.Tree.allowDrop (tree.component.ts:57:19)
    
    You may need a condition check for 
    `dragNode.parent && dragNode.parent.path == [...]`